src/iop/rgbcurve.c

```diff
@@ -1512,17 +1546,17 @@ void cleanup_global(dt_iop_module_so_t *module)
 {
   dt_iop_rgbcurve_global_data_t *gd = (dt_iop_rgbcurve_global_data_t *)module->data;
   dt_opencl_free_kernel(gd->kernel_rgbcurve);
-  free(module->data);
+  dt_free_align(module->data);
   module->data = NULL;
 }
 
-// this will be called from process*()
-// it must be executed only if profile info has changed
-static void _generate_curve_lut(dt_dev_pixelpipe_t *pipe, dt_iop_rgbcurve_data_t *d)
+// called from commit_params
+static void _generate_curve_lut(dt_dev_pixelpipe_t *pipe,
+                                dt_iop_rgbcurve_data_t *d)
 {
   const dt_iop_order_iccprofile_info_t *const work_profile = dt_ioppr_get_pipe_work_profile_info(pipe);
 
-  dt_iop_rgbcurve_node_t curve_nodes[3][DT_IOP_RGBCURVE_MAXNODES];
+  DT_ALIGNED_ARRAY dt_iop_rgbcurve_node_t curve_nodes[3][DT_IOP_RGBCURVE_MAXNODES];
```

Comment:
So after all this and only this is the fix for the issue, right?

Reply:
Yes - i think so too. It's good for all cpus / compilers coming for free.

The "avoiding a possible race condition" by writing the data while committing parameters instead of while `process` is also clearly better.

Let me know if you are ok with this pr or if you want a "stripped down" version for 4.6 only including necessary stuff.

Reply:
@jenshannoschwalm @TurboGit As this is all generic C code with no assumption on the target ISA, isn't this just a workaround for a compiler bug? If the compiler emits code that requires aligned data for an array / a variable that isn't marked as aligned, this looks like over-optimization to me, or did another annotation alure the compiler to assume aligned data?

Sure the code will generally benefit from the alignment, but my point is this being a workaround rather than a solution...

Just my 2¢,
Flössie

Reply:
This is definitely a workaround, and who knows where else in the code we have the same problem.

Reply:
Yes probably a workaround, let's hope the GCC bug (if any) is fixed and we will be safe. At least we stop crashing dt on this part of code and as you said the alignment is a good thing here.

Reply:
> If the compiler emits code that requires aligned data for an array / a variable that isn't marked as aligned, this looks like over-optimization to me

@Floessie It's almost certain that you're right. Would you be willing to invest some time in reporting a bug in the GCC bugtracker?

Reply:
@victoryforce Well, no, as I've lost my login credentials and would have to apply for new ones. And that takes time. I've suggested Hanno to apply for an bugzilla account last week, maybe he has access already.

But... I searched the GCC bugzilla an found an interesting bug, that seems related:

https://gcc.gnu.org/bugzilla/show_bug.cgi?id=110273

Maybe you can chime in there. There is also a very reduced testcase in comment 4 that could be tried on Linux.

HTH and good luck,
Flössie

PS: The GCC folks are very helpful if you manage to provide them with a reduced testcase. In one issue, I was concerned with, the GCC developer even built RawTherapee to reproduce the problem.

Reply:
I didn't yet but will take care of reporting and possibly discussing there.